test/mockupdb/test_mongos_command_read_mode.py

```diff
@@ -61,37 +57,29 @@ def test(self):
         self.addCleanup(server.stop)
         server.run()
         server.autoresponds('ismaster', ismaster=True, msg='isdbgrid',
-                            minWireVersion=2, maxWireVersion=5)
+                            minWireVersion=2, maxWireVersion=6)
 
         pref = make_read_preference(read_pref_mode_from_name(mode),
                                     tag_sets=None)
 
         client = MongoClient(server.uri, read_preference=pref)
         self.addCleanup(client.close)
-        with going(operation.function, client) as future:
-            request = server.receive()
-            request.reply(operation.reply)
+        future = go(operation.function, client)
+        request = server.receive()
+        request.reply(operation.reply)
```

Comment:
What's this change for?

Reply:
I'm still confused. The docs you quoted say that using "going" is more succinct and we also don't use the return value in this test, so why would we change the code to use the more verbose option ("go")?

I think we should change this to use going and remove the explicit "future":
```
        with going(operation.function, client):
            request = server.receive()
            request.reply(operation.reply)
```

Reply:
Done.